components/builder-api/src/server/handlers.rs

```diff
@@ -223,6 +225,73 @@ pub fn get_profile(req: &mut Request) -> IronResult<Response> {
     }
 }
 
+pub fn get_access_tokens(req: &mut Request) -> IronResult<Response> {
+    let session_id = {
+        let session = req.extensions.get::<Authenticated>().unwrap();
+        session.get_id()
+    };
+
+    let mut request = AccountTokensGet::new();
+    request.set_account_id(session_id);
+
+    match route_message::<AccountTokensGet, AccountTokens>(req, &request) {
+        Ok(account_tokens) => Ok(render_json(status::Ok, &account_tokens)),
+        Err(err) => return Ok(render_net_error(&err)),
+    }
+}
+
+pub fn generate_access_token(req: &mut Request) -> IronResult<Response> {
+    let (session_id, flags) = {
+        let session = req.extensions.get::<Authenticated>().unwrap();
+        (session.get_id(), session.get_flags())
+    };
```

Comment:
I was momentarily confused as to why a tuple was used here. I expected `session_id` and `flags` to be used more since they got their own variables, but since they're each referenced only once, I wonder if it might be a bit simpler to eliminate the `let (session_id, flags)` like so:

```rust
    let session = req.extensions.get::<Authenticated>().unwrap();
…
    request.set_id(session.get_id());
…
    let token = bldr_core::access_token::generate_user_token(
        &cfg.depot.key_dir,
        account.get_id(),
        session.get_flags(),
    ).unwrap();
```

Not a major thing, but it makes the code a bit shorter and more evocative to my eye.

Reply:
Yes, that's an unfortunate byproduct of the rust/iron/router combination not being ownership friendly.  If we don't wrap session in it's own scope, we'll get this error, because later on in the code the message routing we're passing the req mutably.

```
error[E0502]: cannot borrow `*req` as mutable because `req.extensions` is also borrowed as immutable
```

So we use this pattern all over the place in the code.

Reply:
Ah, that makes a lot of sense. I should've waited for my build to get this error before I made my suggestion 😬